x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/hooks/setup_technology.ts

```diff
@@ -94,18 +98,43 @@ export function useSetupTechnology({
     isAgentlessCloudEnabled,
     isAgentlessServerlessEnabled,
   } = useAgentless();
+
   // this is a placeholder for the new agent-BASED policy that will be used when the user switches from agentless to agent-based and back
-  const [newAgentBasedPolicy] = useState<NewAgentPolicy | undefined>({ ...newAgentPolicy });
+  const newAgentBasedPolicy = useRef<NewAgentPolicy>(newAgentPolicy);
   const [selectedSetupTechnology, setSelectedSetupTechnology] = useState<SetupTechnology>(
     SetupTechnology.AGENT_BASED
   );
   const [newAgentlessPolicy, setNewAgentlessPolicy] = useState<AgentPolicy | NewAgentPolicy>(
     generateNewAgentPolicyWithDefaults({
-      name: `Agentless policy ${uuidv4()}`,
       supports_agentless: true,
     })
   );
 
+  useEffect(() => {
+    if (isEditPage) {
+      return;
+    }
+    if (isAgentlessCloudEnabled && selectedSetupTechnology === SetupTechnology.AGENTLESS) {
```

Comment:
what happens when a user toggles back and forth from agentless setup to agent-based setup? does the new agent policy name being suggested is reset to the original value? or it is the agentless generated name?

Reply:
It reverts to the original value stored in the `useRef`:

https://github.com/elastic/kibana/assets/19270322/0e118b40-f250-4933-ac92-53d2b1c84038

As can be seen in the video, for simplicity it does not track updates done on the agent-based and then switch to agentless and agent-based again, but that can also be implemented with extra logic if needed.